File: clients/javascript/src/models/topicConfigureIn.ts
// this file is @generated





export interface TopicConfigureIn {
    name: string;
partitions: number;
topic: string;
}

export const TopicConfigureInSerializer = {
    _fromJsonObject(object: any): TopicConfigureIn {
        return {
            name: object['name'],
            partitions: object['partitions'],
            topic: object['topic'],
            };
    },

    _toJsonObject(self: TopicConfigureIn): any {
        return {
            'name': self.name,
            'partitions': self.partitions,
            'topic': self.topic,
            };
    }
}

File: clients/rust/src/models/topic_configure_in.rs
// this file is @generated
use serde::{Deserialize, Serialize};

#[derive(Clone, Debug, Deserialize, Serialize)]
pub struct TopicConfigureIn {
    pub name: String,

    pub partitions: u16,

    pub topic: String,
}

impl TopicConfigureIn {
    pub fn new(name: String, partitions: u16, topic: String) -> Self {
        Self {
            name,
            partitions,
            topic,
        }
    }
}
